astropy/coordinates/representation.py

```diff
@@ -223,6 +224,27 @@ def _unitstr(self):
                            for component in self.components]))
         return unitstr
 
+    def __eq__(self, other):
+        """
+        Check for equality with another representation.
+        """
+        try:
+            comparisons = [getattr(self, component) ==
+                           getattr(other, component)
+                           for component in self.components]
+        except:
```

Comment:
I thought for `__eq__` one should always return something, never raise an `Exception`, although I guess in this particular case the only thing one might be interested in is an `AttributeError`.